src/AllValidators.js

```diff
@@ -26,14 +27,45 @@ export default class AllValidators extends Component {
         for (let i = 0; i < data.length; i++) {
           data[i].index = i + 1
         }
+        return this.augmentValidatorsWithPoPAAddress(data)
+      })
+      .then(augmentedValidators => {
         this.setState({
-          validators: data,
+          validators: augmentedValidators,
           loading: false,
           reload: false,
           netId
         })
       })
   }
+  async augmentValidatorsWithPoPAAddress(validators) {
+    const popa = this.getProofOfPhysicalAddressContract()
+    if (popa === null) {
+      return validators
+    }
+    // Until everything is done ok, result will be validators
+    let result = validators
+    try {
```

Comment:
Added a `try/catch` block here: if something goes wrong, the original `validators` array should be returned and not the augmented one.